### PR TITLE
#167806860 User can edit comment

### PR DESCRIPTION
**Pull request by VictorMaria**
#### What does this PR do?
User can edit their own comment on article

#### Description of Task to be completed?
Has /api/v1/comments/${commentId} working
Add swagger documentation

API Flowchart for edit comment (1)

#### How should this be manually tested?
1. Fetch this branch on the terminal using 
git fetch origin ft-edit-comment-167806860
2. Install dependencies using npm install
3. Start the server with npm run server
4. On Postman, sign in as a verified user by making a POST request to /api/v1/auth/signin and use the token received for the request in the next step.
5. Call /api/v1/comments/${commentId} 

#### Note
You can only edit your own existing comment. Also ensure you enter a valid input in the body of the request

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#167806860](https://www.pivotaltracker.com/story/show/167806860)

#### Screenshots (if appropriate)
Screen Shot 2019-08-12 at 6 24 30 PM

### CODE REVIEW

File: controllers/CommentController.js [hunk 1]
```diff
@@ -124,4 +124,41 @@ export default class CommentController {
...
+    try {
+      const { id } = req.user;
+      const { commentId } = req.params;
+      const { body } = req.body;
+      const comment = await Comment.findOne({ where: { id: commentId } });
+      if (!comment) {
+        return errorResponse(res, 404, 'That comment does not exist');
+      }
+      if (comment.dataValues.userId !== id) {
+        return errorResponse(res, 403, 'You are not allowed to edit another user\'s comment');
```
Comment (ocranbillions) [inline controllers/CommentController.js:161]: Good job checking if the user is the owner of the comment.

File: __tests__/editComment.test.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+  it('should return an error response when a user tries to edit another user\'s comment', (done) => {
+    chai
+      .request(app)
+      .patch(`${url}/comments/${commentId}`)
+      .set('Authorization', `Bearer ${thirdUserToken}`)
+      .send({ body: 'Stop that' })
+      .end((err, res) => {
+        expect(res.status).to.eql(403);
+        expect(res.body).to.have.property('errors');
+        expect(res.body.errors).to.eql('You are not allowed to edit another user\'s comment');
```
Comment (ocranbillions) [inline __tests__/editComment.test.js:112]: Nice!

**Review by ocranbillions — APPROVED** [review]
Good job bro. Kindly document your feature and push again.

**Review by tolulope-od — APPROVED** [review]
LGTM, please check your flowchart, it seems to point that the request will give a server error once it's made

File: __tests__/editComment.test.js (hunk 2 above)
Comment (VictorMaria) [reply __tests__/editComment.test.js:112]: Thank you

**Comment by VictorMaria** [comment]
> LGTM, please check your flowchart, it seems to point that the request will give a server error once it's made

Yeah, That part was tricky for me. I will look into it. Thank you.

**Review by Tunji17 — APPROVED** [review]
LGTM

**Comment by VictorMaria** [comment]
> LGTM

Thank you

**Comment by profmcdan** [comment]
Why are you returning a 403 for invalid token? @VictorMaria

**Review by profmcdan — CHANGES_REQUESTED** [review]
You need to return the right status code for an invalid token, from the flowchart. 
Also, ensure that your code implementation follows the same approach.

**Comment by VictorMaria** [comment]
> Why are you returning a 403 for invalid token? @VictorMaria

It has been corrected. My feature strictly follows the flowchart

**Review by profmcdan — APPROVED** [review]

**Review by profmcdan — CHANGES_REQUESTED** [review]
From the diagram that I can see on your PR, I can still see a 403 being returned. What exactly did you fix?

**Comment by VictorMaria** [comment]
@profmcdan I have uploaded the correct flowchart again

File: docs/comments/editComment.json [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+          }
+        },
+        {
+            "name": "body",
+            "in": "body",
+            "required": "true",
+            "description": "body of the comment",
+            "schema": {
+                "type": "string",
+                "example": "nice one"
```
Comment (OdunayoOkebunmi) [inline docs/comments/editComment.json:29]: Good job. But this returns the "body" as a string, not an object

**Review by profmcdan — APPROVED** [review]

**Pull request merged**